README.md

```diff
@@ -1,91 +1,134 @@
+#include <cstddef>
+#include <string>
 
-Tiny example:
+#include "re2/re2.h"
 
-Assume that you have a `fuzz_test.cc` file to do the fuzzing test and corpus files `corpus_1.txt` and `corpus_dir/*`.
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+    RE2 re(std::string(reinterpret_cast<const char*>(data), size), RE2::Quiet);
+    return 0;
+}
+```
 
-You can create a fuzz test target in the `BUILD` like below:
+### Building and running
 
-```python
-load("@rules_fuzzing//fuzzing:cc_deps.bzl", "cc_fuzz_test")
+To build a fuzz test, you need to specify which fuzzing engine and what instrumentation to use for tracking errors during the execution of the fuzzer. Let's build the RE2 fuzz test using [libFuzzer](https://llvm.org/docs/LibFuzzer.html) and the [Address Sanitizer (ASAN)](https://clang.llvm.org/docs/AddressSanitizer.html) instrumentation, which catches memory errors such as buffer overflows and use-after-frees:
 
-cc_fuzz_test(
-    name = "fuzz_test",
-    srcs = ["fuzz_test.cc"],
-    corpus = ["corpus_1.txt"] + glob(["corpus_dir/**"],
-)
+```sh
+$ bazel build -c opt --config=asan-libfuzzer //examples:re2_fuzz_test
 ```
 
-To run the fuzz target, use the following command:
+You can directly invoke this fuzz test executable if you know libFuzzer's command line interface. But in practice, you don't have to. For each fuzz test `<name>`, the rules library generates a number of additional targets that provide higher-level functionality to simplify the interaction with the fuzz test.
+
+One such target is `<name>_run`, which provides a simple engine-agnostic interface for invoking fuzz tests. Let's run our libFuzzer example:
 
 ```sh
-$ bazel run fuzz_test_run --config=asan-libfuzzer
+$ bazel run -c opt --config=asan-libfuzzer //examples:re2_fuzz_test_run
 ```
 
-You can also control the fuzzing test running time by passing `--timeout_secs` like
+The fuzz test will start running locally, and write the generated tests under a temporary path under `/tmp/fuzzing`. By default, the generated tests persist across runs, in order to make it easy to stop and resume runs (possibly under different engines and configurations).
+
+Let's interrupt the fuzz test execution (Ctrl-C), and resume it using the Honggfuzz engine:
 
 ```sh
-$ bazel run fuzz_test_run --config=asan-libfuzzer -- --timeout_secs=20
+$ bazel run -c opt --config=asan-honggfuzz //examples:re2_fuzz_test_run
 ```
 
-If you only want to run the regression test on the corpus, set `--regression`:
+The `<name>_run` target accepts a number of engine-agnostic flags. For example, the following command runs the fuzz test with an execution timeout and on a clean slate (removing any previously generated tests). Note the extra `--` separator between Bazel's own flags and the launcher flags:
 
 ```sh
-$ bazel run fuzz_test_run --config=asan-libfuzzer -- --regression=True
+$ bazel run -c opt --config=asan-libfuzzer //examples:re2_fuzz_test_run -- --clean --timeout_secs=30
```

Comment:
nit: Maybe break to new line \
-- --clean --timeout_secs30
to make it more obvious ?

Reply:
Done.